app/assets/javascripts/discourse/app/components/user-menu/notifications-list.js

```diff
@@ -0,0 +1,74 @@
+import UserMenuItemsList from "discourse/components/user-menu/items-list";
+import I18n from "I18n";
+import { action } from "@ember/object";
+
+export default class UserMenuNotificationsList extends UserMenuItemsList {
+  get filterByTypes() {
+    return null;
+  }
+
+  get showAll() {
+    return true;
+  }
+
+  get showAllHref() {
+    return `${this.currentUser.path}/notifications`;
+  }
+
+  get showAllTitle() {
+    return I18n.t("user_menu.view_all_notifications");
+  }
+
+  get showDismiss() {
+    return this.items.some((item) => !item.read);
+  }
+
+  get dismissTitle() {
+    return I18n.t("user.dismiss_notifications_tooltip");
+  }
+
+  get itemsCacheKey() {
+    let key = "recent-notifications";
+    const types = this.filterByTypes?.toString();
+    if (types) {
+      key += `-type-${types}`;
+    }
+    return key;
+  }
+
+  get emptyStateComponent() {
+    if (this.constructor === UserMenuNotificationsList) {
```

Comment:
Not sure if I"m missing something but wouldn't the constructor always be `UserMenuNotificationsList` here?

Reply:
Ah I guess another component might inherit from this here?

Reply:
I kind of prefer if the component inheriting can just override this function completely if it needs a different empty state component.

Reply:
yes, there are a few components that inherit from `notifications-list` and I don't want the children to get `notifications-list`'s empty state component. I think in the future each child of `notifications-list` will have its own empty state component so this is kind of temporary.